app/docker/services/serviceService.js

```diff
@@ -1,8 +1,23 @@
 angular.module('portainer.docker')
-.factory('ServiceService', ['$q', 'Service', 'ServiceHelper', 'TaskService', 'ResourceControlService', function ServiceServiceFactory($q, Service, ServiceHelper, TaskService, ResourceControlService) {
+.factory('ServiceService', ['$q', 'Service', 'ServiceHelper', 'TaskService', 'ResourceControlService', 'Notifications', function ServiceServiceFactory($q, Service, ServiceHelper, TaskService, ResourceControlService, Notifications) {
   'use strict';
   var service = {};
 
+  service.scaleService = function(selectedService, $state) {
```

Comment:
I do not think that the introduction of this method is coherent with how the `ServiceService` is supposed to work.

Only ngresource/promise handling is usually done in service. While this refactor and centralize the scaling function, it also introduce components such as `Notifications` and `$state` which should not be part of this service.

Please remove it.

Reply:
ok..I will take a lot.. I was trying to avoid duplicate code.

Reply:
I know but I do not think this is the best way to refactor and centralize this function.

We might want to introduce another service layer later on, for now it will be fine.

Reply:
no prob..I will post another pull request today.